doc/tutorials/models/model_parameters/main.md

```diff
@@ -0,0 +1,202 @@
+@insert-title:tutorials-models-model-parameters
+
+[TOC]
+
+## Problem description
+
+In this tutorial, we will revisit the problem defined in the previous tutorial and demonstrate how to work with model parameters.
+
+Recall that the linear elasticity material model can be mathematically written as
+\f[
+  \boldsymbol{\sigma} = 3 K \operatorname{vol} \boldsymbol{\varepsilon} + 2 G \operatorname{dev} \boldsymbol{\varepsilon}.
+\f]
+Also recall that all NEML2 models can be written in the following general form
+\f[
+  y = f(x; p, b).
+\f]
+Pattern matching suggests the following set definitions:
+\f[
+  x = \left\{ \boldsymbol{\varepsilon} \right\}, \quad y = \left\{ \boldsymbol{\sigma} \right\}, \quad p = \left\{ K, G \right\}, \quad b = \varnothing.
+\f]
+
+## Parameter v.s. buffer
+
+Both \f$ K \f$ and \f$ G \f$ are here categorized as model parameters. The major differences between parameters and buffers are
+- Parameters are "trainable", whereas buffers are not. NEML2 can use automatic differentiation to calculate the derivative of output variables w.r.t. the model parameters.
+- Parameters can be (recursively) defined by other models, whereas buffers cannot.
```

Comment:
You may want to clarify this with an example.

Reply:
I added a link here pointing to "Model parameters revisited" which should explain this.